flax/nnx/nnx/bridge/wrappers.py

```diff
@@ -143,16 +144,17 @@ def __call__(
       if 'params' not in _rngs and 'default' in _rngs:
         _rngs['params'] = _rngs.pop('default')
       out, variables = self.module.init_with_output(_rngs, *args, method=method, **kwargs)
-      def nn_var_to_nnx_state(kp, v):
-        assert isinstance(kp[0], jtu.DictKey)
-        vtype = variableslib.variable_type(kp[0].key)
-        return vtype(v)
-      for col, tree in jtu.tree_map_with_path(nn_var_to_nnx_state, variables).items():
+
+      nnx_vars = jtu.tree_map_with_path(bv.to_nnx_var, variables,
+                                        is_leaf=lambda x: isinstance(x, meta.AxisMetadata))
+      for col, tree in nnx_vars.items():
         self._setattr(col, tree)
```

Comment:
A bit more idiomatic:
```suggestion
        setattr(self, col, tree)
```

Reply:
Done.